runtime/src/main/java/io/micronaut/jackson/serialize/OptionalValuesSerializer.java

```diff
@@ -64,4 +73,8 @@ public void serialize(OptionalValues<?> value, JsonGenerator gen, SerializerProv
         }
         gen.writeEndObject();
     }
+
+    private boolean canSerializeElementAsObject(Class<?> type) {
+        return type != JsonError.class || !jacksonConfiguration.getHateoas().isAlwaysSerializeErrorsAsList();
```

Comment:
Please assign `jacksonConfiguration.getHateoas().isAlwaysSerializeErrorsAsList()` to a final local field to avoid repeated method calls